rb_tree.py

```diff
@@ -1,81 +1,287 @@
+# -*- coding: utf-8 -*-
 """
-A classic (not left-leaning) Red-Black Tree implementation, supporting addition and deletion.
+A classic (not left-leaning) Red-Black Tree implementation, supporting 
+addition and deletion. The code is by Stanislav Kozlovski, modified
+slightly by George Flanagin of University of Richmond, with the purpose
+of having it look somewhat more like the rest of the code at the
+university. The changes are only additions.
+
+ 1. The color of the nodes is now an Enum.
```

Comment:
I'd prefer we keep the changes outside the source file itself. This is a very good description for the git commit message and PR description.